### PR TITLE
bug in RemotePointEvaluation::point_found(i) for i=0

### PR DESCRIPTION
**Pull request by nfehn**
I think there is a bug in the code, because for `i=0`, we access `point_ptrs[0-1]`???

I don't know whether the logic I implemented is correct. I can only state that it avoids the segmentation fault.

### CODE REVIEW

File: source/base/mpi_remote_point_evaluation.cc [hunk 1]
```diff
@@ -207,7 +207,7 @@ namespace Utilities
       const unsigned int i) const
     {
       AssertIndexRange(i, point_ptrs.size());
-      return (point_ptrs[i] - point_ptrs[i - 1]) > 0;
+      return (point_ptrs[i] - (i > 0) ? point_ptrs[i - 1] : 0) > 0;
```
Comment (peterrum) [inline source/base/mpi_remote_point_evaluation.cc:210]:
```suggestion
      return (point_ptrs[i+1] - point_ptrs[i]) > 0;
```

Comment (nfehn) [reply source/base/mpi_remote_point_evaluation.cc:210]: Just to be sure: Is it then correct that `all_points_found()` begins the loop with `i=1`?

Comment (peterrum) [reply source/base/mpi_remote_point_evaluation.cc:210]: Yes it is right.

Comment (nfehn) [reply source/base/mpi_remote_point_evaluation.cc:210]: And the AssertIndexRange ??

Comment (peterrum) [reply source/base/mpi_remote_point_evaluation.cc:210]: `AssertIndexRange(i, point_ptrs.size() - 1);`

Comment (nfehn) [reply source/base/mpi_remote_point_evaluation.cc:210]: But then the loop in `all_points_found()` can not run its last element, right?

Comment (peterrum) [reply source/base/mpi_remote_point_evaluation.cc:210]: Now I understand what you mean! `all_points_found()` calls this function. It has to loop from 0 to `point_ptrs.size() - 1`. Please add an if-statement around the loop `if(point_ptrs.size() > 0)`.

**Comment by nfehn** [comment]
please review again

**Review by kronbichler — APPROVED** [review]

**Review by peterrum — APPROVED** [review]
Thanks!

**Pull request merged**